src/main/scala/Configs.scala

```diff
@@ -61,87 +61,103 @@ class WithNBlockDeviceTrackers(n: Int) extends Config((site, here, up) => {
-  case SerialTLKey => up(SerialTLKey).map(s => s.copy(attachParams=s.attachParams.copy(masterWhere = masterWhere)))
+  case SerialTLKey => up(SerialTLKey).map(s => s.copy(
+    client=s.client.map(_.copy(masterWhere=masterWhere))
+  ))
 })
 
 class WithSerialTLSlaveLocation(slaveWhere: TLBusWrapperLocation) extends Config((site, here, up) => {
-  case SerialTLKey => up(SerialTLKey).map(s => s.copy(attachParams=s.attachParams.copy(slaveWhere = slaveWhere)))
+  case SerialTLKey => up(SerialTLKey).map(s => s.copy(
+    manager=s.manager.map(_.copy(slaveWhere=slaveWhere))
+  ))
 })
 
 class WithSerialTLPBusManager extends WithSerialTLSlaveLocation(PBUS)
 
-class WithSerialSlaveCrossingType(xType: ClockCrossingType) extends Config((site, here, up) => {
-  case SerialTLKey => up(SerialTLKey).map(s => s.copy(attachParams=s.attachParams.copy(slaveCrossingType = xType)))
+class WithSerialTLClient extends Config((site, here, up) => {
+  case SerialTLKey => up(SerialTLKey).map(s => s.copy(client=Some(SerialTLClientParams())))
 })
 
-class WithAsynchronousSerialSlaveCrossing extends WithSerialSlaveCrossingType(AsynchronousCrossing())
+class WithNoSerialTLClient extends Config((site, here, up) => {
+  case SerialTLKey => up(SerialTLKey).map(s => s.copy(client=None))
+})
 
 class WithSerialTLMem(
   base: BigInt = BigInt("80000000", 16),
   size: BigInt = BigInt("10000000", 16),
   idBits: Int = 8,
-  isMainMemory: Boolean = true,
-  bundleParams: TLBundleParameters = TLSerdesser.STANDARD_TLBUNDLE_PARAMS
+  isMainMemory: Boolean = true
 ) extends Config((site, here, up) => {
   case SerialTLKey => {
-    val masterPortParams = MasterPortParams(
-      base = base,
-      size = size,
-      idBits = idBits,
-      beatBytes = site(MemoryBusKey).beatBytes
+    val memParams = ManagerRAMParams(
+      address = base,
+      size = size
     )
     up(SerialTLKey, site).map { k => k.copy(
-      serialTLManagerParams = Some(k.serialTLManagerParams.getOrElse(SerialTLManagerParams(memParams = masterPortParams))
-        .copy(memParams = masterPortParams, isMemoryDevice = isMainMemory)),
-      bundleParams = bundleParams
+      manager = Some(k.manager.getOrElse(SerialTLManagerParams()).copy(
+        memParams = Seq(memParams),
+        isMemoryDevice = isMainMemory,
+        idBits = idBits
+      ))
     )}
   }
 })
 
+class WithSerialTLBundleParams(params: TLBundleParameters) extends Config((site, here, up) => {
+  case SerialTLKey => up(SerialTLKey).map(_.copy(bundleParams=params))
+})
 
-class WithSerialTLBackingMemory extends Config((site, here, up) => {
-  case ExtMem => None
-  case SerialTLKey => up(SerialTLKey, site).map { k => k.copy(
-    serialTLManagerParams = Some(k.serialTLManagerParams.getOrElse(SerialTLManagerParams(memParams = up(ExtMem).get.master))
-      .copy(memParams = up(ExtMem).get.master, isMemoryDevice = true))
-  )}
+class WithSerialTLBCE extends Config((site, here, up) => {
```

Comment:
One linear please. Better not leave it ambiguous

Reply:
You mean you prefer this as a one-liner?

Reply:
I meant to add a one line comment

Reply:
Ok, I did that in the most recent commits.